javaslang/src/main/java/javaslang/collection/Seq.java

```diff
@@ -1153,4 +1149,38 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
     default Function1<Integer, T> withDefault(Function<? super Integer, ? extends T> defaultFunction) {
         return i -> (i >= 0 && i < length()) ? apply(i) : defaultFunction.apply(i);
     }
+
+    default java.util.List<T> asUnmodifiableList() { return new SeqList<>(Seq.this); }
```

Comment:
if we use `unmodifiable`, I don't think we need `Java`, since it's obviously referring to the `Collections'` method, as you also pointed out.
If we use immutable, we may need `Java` in the name also.

> or one thing, they aren't really immutable

Thanks to reflection and `UNSAFE`, nothing really is :)